client/components/OverlayManager.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useRef, useEffect } from 'react';
```

Comment:
As we had discussed, I think it makes sense to have an `OverlayManager` that handles all of the fullscreen popups, and some "HoC" (which, with hooks, I think is a custom hook) that handles the z-index and click-outside-to-close functionality.

Reply:
I think this is worth a refactor in a new branch. What do you think?

Reply:
In hindsight, this is kind of crucial to get the css refactor done, so I chose to do it now

Reply:
So, I came up with a pretty neat solution for this: a function called `useAsModal` (defined in `custom-hooks.js` - btw, it needs a better place). You pass a `<Component />` to this function, and it returns `[trigger, WrappedComponent]`, where:
- trigger is a callback to display the component
-  `<WrappedComponent />` is the component, with modal behavior

This *might* be an antipattern, because the hook returns jsx (though there's no consensus on this), but it's the most practical/versatile solution I could find.